test/functional/tests.py

```diff
@@ -807,6 +807,68 @@ def test_bitswap_unwant(self):
         result = self.api.bitswap_unwant(key='QmZTR5bcpQD7cFgTorqxZDYaew1Wqgfbd2ud9QqGPAkK2V')
         self.assertTrue(result is not None)
 
+@skipIfOffline()
+class IpfsApiPubSubTest(unittest.TestCase):
+
+    def setUp(self):
+        self.api = ipfsapi.Client()
+
+    def test_pubsub_pub_sub(self):
+        """
+        We should test both publishing and subscribing
+        at the same time for it's difficult
+        to determine if one works without checking the
+        other.
+        """
+        # the topic that will be published/subscribed to
+        topic = 'testing'
```

Comment:
Please use an unique channel topic like per session (for instance by using uuid like in `"{}.testing".format(uuid.uuid4())`): As far as I know pubsub messages are supposed to be impliciently propagated across subscribing nodes in a future version and with these generic names there'd be a high chance of random interference from other nodes (and also interferance from other people possibly running these same unit tests) which would cause “random errors in tests” – something that everybody hates to debug.

(Also in the other test below.)